ts/store/reducers/featureFlagWithMinAppVersionStatus.ts

```diff
@@ -0,0 +1,124 @@
+import { Platform } from "react-native";
+import { pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import * as E from "fp-ts/lib/Either";
+import { PatternString } from "@pagopa/ts-commons/lib/strings";
+import { BackendStatus } from "../../../definitions/content/BackendStatus";
+import { Config } from "../../../definitions/content/Config";
+import { VersionPerPlatform } from "../../../definitions/content/VersionPerPlatform";
+import { getAppVersion, isVersionSupported } from "../../utils/appVersion";
+
+type ObjectWithMinAppVersion =
+  | { min_app_version?: VersionPerPlatform }
+  | undefined;
+
+// This type extracts all keys that have a structure with min_app_version
+type KeysWithMinAppVersion<T> = Extract<
+  keyof T,
+  {
+    [K in keyof T]: T[K] extends ObjectWithMinAppVersion ? K : never;
+  }[keyof T]
+>;
+
+// This type extracts all keys that have a structure with min_app_version, nested in another type
+type ExtractSecondLevelKeyWithMinAppVersion<
+  T,
+  FirstLevel extends keyof T
+> = T[FirstLevel] extends infer U
+  ? U extends object
+    ? KeysWithMinAppVersion<U>
+    : never
+  : never;
+
+// This type defines the parameters for a function so that the name of the parameter must
+// be specified and that any optional configuration of a FF is present in whole, not in part
+type CheckPropertyWithMinAppVersionParameters<
+  T extends KeysWithMinAppVersion<Config>
+> = {
+  backendStatus: O.Option<BackendStatus>;
+  mainLocalFlag: boolean;
+  configPropertyName: T;
+} & (
+  | { optionalConfig?: undefined; optionalLocalFlag?: undefined }
+  | {
+      optionalLocalFlag: boolean;
+      optionalConfig: ExtractSecondLevelKeyWithMinAppVersion<Config, T>;
+    }
+);
+
+// This feature checks that a feature flag is enabled by checking the local option and the minimum
+// version of the feature set remotely.
+// It is possible to specify an optional configuration that corresponds to a feature flag nested into the main one.
+// If the main FF is deactivated, any nested FF will also be considered deactivated.
```

Comment:
Maybe this function deserves a more JS Doc comment, with a detail description. What do you think?